libr/asm/asm.c

```diff
@@ -65,9 +65,11 @@ static inline int r_asm_pseudo_org(RAsm *a, char *input) {
 
 // wtf isnt this the same as r_asm_op_set_hex() ??
 static inline int r_asm_pseudo_hex(RAsmOp *op, char *input) {
-	// assume input fits inside op->buf
-	int len = r_hex_str2bin (input, (ut8*)r_strbuf_get (&op->buf));
-	r_asm_op_set_hex (op, r_str_trim_head_tail (input));
+    	ut8 *opBuf = r_asm_op_get_buf (op);
+	// Truncate the input based on the size of the opBuf
+    	input = r_str_newlen (r_str_trim_head_tail (input), sizeof (RStrBuf));
+    	int len = r_hex_str2bin (input, opBuf);
+    	r_asm_op_set_hex (op, input);
```

Comment:
r_asm_op_set_bin() is currently not present in the api.
But like you said r_asm_op_set_hex is implicitly setting the bin.
Also r_hex_str2bin is being used to get the length so that we can throw an error and return it.
r_asm_op_set_hex() api does not return the length. I didn't want to modify it since it might affect the other usages of that method.